### ISSUE TITLE
Streaming responses not working on Blazor WASM

### ISSUE THREAD

**Issue by mili-tan** [awaescher/OllamaSharp#84]
Hi there
I found a problem on Blazor WASM, usually by default the streaming response doesn't work. The following needs to be added to make it work:

```
	private async IAsyncEnumerable<TResponse?> StreamPostAsync<TRequest, TResponse>(string endpoint, TRequest ollamaRequest, [EnumeratorCancellation] CancellationToken cancellationToken) where TRequest : OllamaRequest
	{
		var requestMessage = new HttpRequestMessage(HttpMethod.Post, endpoint)
		{
			Content = new StringContent(JsonSerializer.Serialize(ollamaRequest, OutgoingJsonSerializerOptions), Encoding.UTF8, "application/json")
		};

		//////////
		requestMessage.SetBrowserResponseStreamingEnabled(true);
		//////////

		using var response = await SendToOllamaAsync(requestMessage, ollamaRequest, HttpCompletionOption.ResponseHeadersRead, cancellationToken);

		await foreach (var result in ProcessStreamedResponseAsync<TResponse>(response, cancellationToken))
			yield return result;
	}
```
But it depends on `Microsoft.AspNetCore.Components.WebAssembly`, which may add many unnecessary dependencies for other platforms.
Maybe we should add a new branch and nuget package for Blazor WASM, or is there another good way to do this?

**Comment by awaescher** [comment]
Whoa, I hate this already. I guess we should encourage to use a derived `OllamaApiClient` (say `WasmOllamaApiClient`) and allow to override the creation of new webrequests and maybe `SendToOllamaAsync()` so they can do whatever they need with the requests.

**Comment by awaescher** [comment]
Fixes with #87, thanks mili-tan
